### PR TITLE
Only add parameters to the request object if they are specified

### PR DESCRIPTION
**Pull request by Coenego**
When sending a request to the REST API, we should only add the body parameters if they have been explicitly specified.

For example:
```
someFunc('', 'someValue');

/**
 * @param  {String}    [valueA]
 * @param  {String}    valueB
 */
someFunc(valueA, valueB) {

    // The request body
    var requestBody = {
        'valueB': valueB 
    };

    // Only add valueA if it has been explicitly been specified
    if (valueA) {
        requestBody.valueA = valueA;
    }

    $.ajax({
        'data': requestBody
    });
};
```

### CODE REVIEW

**Comment by Coenego** [comment]
Assigning to @bp323 for review

File: shared/gh/api/gh.api.event.js [hunk 1]
```diff
@@ -153,22 +172,48 @@ define(['exports'], function(exports) {
...
+        // Request data object
+        var data = {
+            'app': appId,
+            'displayName': displayName,
+            'start': start,
+            'end': end
+        };
+
+        // Only add the parameters to the request object if they have been explicitly specified
+        /* istanbul ignore next */
```
Comment (bp323) [inline shared/gh/api/gh.api.event.js]: Can't we test these quite easily?

**Comment by bp323** [comment]
Looks good, one question remaining though. I'll also hold off merging until the build is stable again.

**Comment by Coenego** [comment]
Reassigning to @bp323 for review after follow-up

**Comment by bp323** [comment]
Thanks for going through this, it's a pain :f Verified and merged!

**Pull request merged**